src/manager.cpp

```diff
@@ -212,12 +212,194 @@ void Manager::SetTimerToDetectVpdCollectionStatus()
 int Manager::updateKeyword(const types::Path i_vpdPath,
                            const types::WriteVpdParams i_paramsToWriteData)
 {
-    // Code to supress unused variable warning. To be removed.
-    (void)i_vpdPath;
-    (void)i_paramsToWriteData;
+    if (i_vpdPath.empty())
+    {
+        logging::logMessage("Please provide a VPD path.");
+        return -1;
+    }
 
-    // On success return number of bytes written. On failure return -1.
-    return -1;
+    nlohmann::json l_sysCfgJsonObj{};
+
+    if (m_worker.get() != nullptr)
+    {
+        l_sysCfgJsonObj = m_worker->getSysCfgJsonObj();
+    }
+
+    // If system config JSON object is empty
+    if (l_sysCfgJsonObj.empty())
+    {
+        // Perform write on hardware on the given path
+        return (
+            writeOnHardware(i_vpdPath, l_sysCfgJsonObj, i_paramsToWriteData));
+    }
+
+    int l_bytesWritten = -1;
+
+    try
+    {
+        // For the given VPD path, get it's corresponding inventory object path
+        // from system config JSON
+        const std::string& l_inventoryObjPath =
+            jsonUtility::getInventoryObjPathFromJson(l_sysCfgJsonObj,
+                                                     i_vpdPath);
+
+        // For the given VPD path, get it's corresponding hardware path from
+        // system config JSON.
+        const std::string& l_fruPath =
+            jsonUtility::getFruPathFromJson(l_sysCfgJsonObj, i_vpdPath);
+
+        // If given VPD path is not found in JSON
+        if (l_inventoryObjPath.empty() && l_fruPath.empty())
+        {
+            // Perform write on hardware on the given path
+            return (writeOnHardware(i_vpdPath, l_sysCfgJsonObj,
+                                    i_paramsToWriteData));
+        }
+
+        // To hold the copy of l_dbusObjMap, which is required incase of
+        // reverting the data from D-bus.
+        types::ObjectMap l_copyOfDbusObjMap;
+
+        types::DbusInterface l_interfaceName;
+        types::DbusPropertyName l_propertyName;
+        types::DbusVariantType l_keywordCurrentData;
+
+        // Extract data from input to write on D-bus
+        if (const types::IpzData* l_ipzData =
```

Comment:
nope, I need to revert the data on D-bus if write fails on hardware. As part of which I need the dbus object map, keyword's actual value before performing write. So again I need to have all those logic in this updateKeyword() api if I have a separate api for writeOnDbus. So I prefer not to have another api to writeOnDbus, as it will not find any purpose.

Reply:
Sorry, I did not understand. Why can't all these extraction logic be inside Write Dbus API itself. That will put less noise into a single API and a separate functionality will be extracted.
Please try to see if there is a design through which it can be achieved.

For example, one way could be, API can be designed in a way such as, just the updated input parameter can be passed to that API and that API takes care of writing data  on D-Bus?
or, any other better way that you can think about.